netenv/online-status.go

```diff
@@ -200,13 +200,18 @@ func updateOnlineStatus(status OnlineStatus, portalURL *url.URL, comment string)
 
 	// trigger event
 	if changed {
-		module.TriggerEvent(OnlineStatusChangedEvent, nil)
+		module.TriggerEvent(OnlineStatusChangedEvent, status)
 		if status == StatusPortal {
 			log.Infof(`netenv: setting online status to %s at "%s" (%s)`, status, portalURL, comment)
 		} else {
 			log.Infof("netenv: setting online status to %s (%s)", status, comment)
 		}
 		triggerNetworkChangeCheck()
+
+		// Trigger update check when coming (semi) online.
+		if Online() {
+			_ = updates.TriggerUpdate(false)
+		}
 	}
```

Comment:
Hm, I guess it depends but should'nt the updates module better listen to `OnlineStatusChangedEvent` instead of adding a new dependency on `updates` to `netenv`?

Reply:
I tried listening to the event, but cannot get the online status, because of a dependency loop. Because the `OnlineStatus` is a separate type, I cannot even manually cast it when it is supplied as event data.